cmd/flux/cmd/fmt.go

```diff
@@ -0,0 +1,65 @@
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/influxdata/flux/ast"
+	"github.com/influxdata/flux/ast/astutil"
+	"github.com/influxdata/flux/parser"
+	"github.com/spf13/cobra"
+)
+
+// fmtCmd represents the fmt command
+var fmtCmd = &cobra.Command{
+	Use:   "fmt",
+	Short: "Format a Flux script-file",
+	Long:  "Format a Flux script-file (flux fmt [-w] <script-file>)",
+	Args:  cobra.MinimumNArgs(1),
+	RunE:  format,
+}
+
+var writeResultToSource bool
+
+func init() {
+	rootCmd.AddCommand(fmtCmd)
+	fmtCmd.Flags().BoolVarP(&writeResultToSource, "write-result-to-source", "w", false, "write result to (source) file instead of stdout")
+}
+
+func format(cmd *cobra.Command, args []string) error {
+	script := args[0]
+
+	fromFile, err := ioutil.ReadFile(script)
+	if err != nil {
+		return err
+	}
+
+	pkg := parser.ParseSource(string(fromFile))
+	if ast.Check(pkg) > 0 {
+		return fmt.Errorf("unexpected error: %s , filename: %s", ast.GetError(pkg), script)
+	} else if len(pkg.Files) != 1 {
+		return fmt.Errorf("expected one file in the package, got %d", len(pkg.Files))
+	}
+
+	formatStr, err := astutil.Format(pkg.Files[0])
+	if err != nil {
+		return fmt.Errorf("failed to format the query: %v, filename: %s", err, script)
+	}
+
+	if writeResultToSource {
+		if string(fromFile) != formatStr {
+			return updateScript(script, formatStr)
+		}
+	} else {
+		fmt.Printf("Formatted script:\n\n%s\n\n", formatStr)
```

Comment:
If the formatted file gets written to `stdout`, won't this line interfere with the output? It would be good to write to `stderr` here instead I think.

Also, it might look nice if this output was just one line (without the new lines) since especially in CI we may be formatting many files at a time.

Reply:
I will remove that line thanks.